undefined reference to `meminfo'

ubuntu 14.04
 at firest:  sudo apt-get install libprocps3-dev
then make~~
g++ -O3 -march=native -lprocps -Wall    -o ballooner main.cpp
/tmp/cc64iw70.o: In function `update_m_info()':
main.cpp:(.text+0x5): undefined reference to`meminfo'
main.cpp:(.text+0xc): undefined reference to `kb_main_cached'
main.cpp:(.text+0x1d): undefined reference to`kb_main_buffers'
main.cpp:(.text+0x2e): undefined reference to `kb_main_free'
/tmp/cc64iw70.o: In function`main':
main.cpp:(.text.startup+0x154): undefined reference to `meminfo'
main.cpp:(.text.startup+0x15b): undefined reference to`kb_main_cached'
main.cpp:(.text.startup+0x162): undefined reference to `kb_main_buffers'
main.cpp:(.text.startup+0x175): undefined reference to`kb_main_free'
collect2: error: ld returned 1 exit status
make: **\* [ballooner] Error 1

--- a/Makefile
+++ b/Makefile
@@ -10,4 +10,4 @@ run: all
 	./ballooner 500
 
 ballooner : main.cpp
-	$(CC) $(CFLAGS)   -o $@ $<
+	$(CC) $< $(CFLAGS) -o $@

